2748.Number of Beautiful Pairs/solution.py

```diff
@@ -0,0 +1,17 @@
+class Solution:
+    # Counts the number of beautiful pairs in a 0-indexed integer array called nums
+    # where a pair of indices i and j is called beautiful when the first digit of 
+    # nums[i] and the last digit of nums[j] are coprime and:
+    # 0 <= i < j < nums.length
+    # 2 <= nums.length <= 100
+    # 1 <= nums[i] <= 9999
+    # nums[i] is not divisible by 10
+    def countBeautifulPairs(self, nums: List[int]) -> int:
+        
+        # initialize a counter that keeps track of how many beautiful pairs there are
+        # loop through each index in the array
+            # for each index's value, compare it with each larger index's value via:
+            # gcd(first, last) where first = int(str(num)[0]) and last = num%10
```

Comment:
These are excellent implementation notes but not as good comments which should seek to explain the code rather than duplicate what you will put into code.

Reply:
I didn't think you could set "last" to num%10, though!

Reply:
I think you should convert all of the numbers to a tuple of its first and last digits, which looks like this:
`[12, 13, 24]` gets converted to `[(1, 2), (1, 3), (2, 4)]`
This is just a comment that will save you a bit of runtime.

Reply:
Processing test case...
All 3 pairs are beautiful. This is not a good test case.

Also, I wrote about that in my total review. Should've written it here, though.